eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -223,58 +224,55 @@ function New-MgmtPackageFolder() {
         [string]$packageName = "",
         [string]$sdkPath = "",
-        [string]$commitid = "",
         [string]$readme = "",
         [string]$AUTOREST_CONFIG_FILE = "autorest.md",
         [string]$outputJsonFile = "newPacakgeOutput.json"
     )
   
-    $projectFolder="$sdkPath/sdk/$packageName/Azure.ResourceManager.*"
+    if ($packageName -eq "") {
+        $packageName = $service
+    }
+    # $projectFolder="$sdkPath/sdk/$packageName/Azure.ResourceManager.*"
+    $projectFolder = (Join-Path $sdkPath "sdk" $packageName "Azure.ResourceManager.*")
+    $mgmtPackageName = ""
+    $projectFolder = $projectFolder -replace "\\", "/"
     if (Test-Path -Path $projectFolder) {
       Write-Host "Path exists!"
       $folderinfo = Get-ChildItem -Path $projectFolder
-      $foldername = $folderinfo.Name
-      $projectFolder = "$sdkPath/sdk/$packageName/$foldername"
+      $mgmtPackageName = $folderinfo.Name
+      $projectFolder = "$sdkPath/sdk/$packageName/$mgmtPackageName"
     } else {
       Write-Host "Path doesn't exist. create template."
       dotnet new -i $sdkPath/eng/templates/Azure.ResourceManager.Template
-      $projectFolder="$sdkPath/sdk/$packageName/Azure.ResourceManager.$packageName"
+      $CaptizedPackageName = (Get-Culture ).TextInfo.ToTitleCase($packageName)
+      $mgmtPackageName = "Azure.ResourceManager.$CaptizedPackageName"
+      $projectFolder="$sdkPath/sdk/$packageName/Azure.ResourceManager.$CaptizedPackageName"
       Write-Host "Create project folder $projectFolder"
       New-Item -Path $projectFolder -ItemType Directory
       Push-Location $projectFolder
       dotnet new azuremgmt --provider $packageName --includeCI true --force
       Pop-Location
     }
   
-    # update the readme url if needed.
-    if ($commitid -ne "") {
+    # update the readme path.
+    if ($readme -ne "") {
       Write-Host "Updating autorest.md file."
-      $swaggerInfo = Get-SwaggerInfo -dir "$projectFolder/src"
-      $org = $swaggerInfo[0]
-      $rp = $swaggerInfo[1]
-      $permalinks = "https://github.com/$org/azure-rest-api-specs/blob/$commitid/specification/$rp/resource-manager/readme.md"
-      $requirefile = "require: $permalinks"
-      $rquirefileRex = "require *:.*.md"
-      $file="$projectFolder/src/$AUTOREST_CONFIG_FILE"
-      (Get-Content $file) -replace $rquirefileRex, "$requirefile" | Set-Content $file
-    } elseif ($readme -ne "") {
-      Write-Host "Updating required file $readme in autorest.md file."
       $requirefile = "require: $readme"
       $rquirefileRex = "require *:.*.md"
       $file="$projectFolder/src/$AUTOREST_CONFIG_FILE"
       (Get-Content $file) -replace $rquirefileRex, "$requirefile" | Set-Content $file
-  
-      $readmefilestr = Get-Content $file
-      Write-Output "autorest.md:$readmefilestr"
     }
   
-    $path=$projectFolder
-    $path=$path.Replace($sdkPath + "/", "")
+    Push-Location $sdkPath
+    $relativeFolderPath = Resolve-Path $projectFolder -Relative
+    Pop-Location
```

Comment:
As noted above, don't abuse push and pop this way. It can cause catastrophic issues.
```suggestion
    $relativeFolderPath = Resolve-Path "$sdkPath/$projectFolder" -Relative
```